src/lib/gif-decoder.js

```diff
@@ -0,0 +1,69 @@
+import {GifReader} from 'omggif';
+
+export default (arrayBuffer, {onFrame, onDone}) => {
+    const canvas = document.createElement('canvas');
+    const ctx = canvas.getContext('2d');
+    const gifReader = new GifReader(new Uint8Array(arrayBuffer));
+    canvas.width = gifReader.width;
+    canvas.height = gifReader.height;
+
+    let imageData = ctx.createImageData(canvas.width, canvas.height);
+    let previousData = ctx.createImageData(canvas.width, canvas.height);
+
+    const loadFrame = i => {
+        const framePixels = [];
+        gifReader.decodeAndBlitFrameRGBA(i, framePixels);
+        const {x, y, width, height, disposal} = gifReader.frameInfo(i);
+
+        for (let row = 0; row < height; row++) {
+            for (let column = 0; column < width; column++) {
+                const indexOffset = 4 * (x + (y * canvas.width));
+                const j = indexOffset + (4 * (column + (row * canvas.width)));
+                if (framePixels[j + 3]) {
+                    imageData.data[j + 0] = framePixels[j + 0];
+                    imageData.data[j + 1] = framePixels[j + 1];
+                    imageData.data[j + 2] = framePixels[j + 2];
+                    imageData.data[j + 3] = framePixels[j + 3];
+                }
+            }
+        }
+
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        ctx.putImageData(imageData, 0, 0);
+
+        const dataUrl = canvas.toDataURL();
+
+        switch (disposal) {
+        case 0:
+        case 1: // Both are "keep previous frame"
+            previousData = ctx.getImageData(0, 0, canvas.width, canvas.height);
+            break;
```

Comment:
This is the default, so this case could be removed and commented about there.

Reply:
@rschamp fixed in 1d617214b4afb8dbf3f973c8f85d5b03f4be50d7